py-polars/polars/io/spreadsheet/functions.py

```diff
@@ -532,7 +571,13 @@ def _read_spreadsheet(
     )
     try:
         # parse data from the indicated sheet(s)
-        sheet_names, return_multi = _get_sheet_names(sheet_id, sheet_name, worksheets)
+        sheet_names, return_multiple_sheets = _get_sheet_names(
+            sheet_id, sheet_name, worksheets
+        )
+        if read_multiple_workbooks and return_multiple_sheets:
+            msg = "cannot return multiple sheets from multiple workbooks"
```

Comment:
In theory, I imagine it's possible that you might have multiple files, each containing multiple sheets of the same names and structure, and you might want to read all sheets from all files into a big dictionary of sheets. What was the reason for making these options mutually exclusive?

Reply:
Two reasons:

1) The chances are significantly higher that this isn't the case.
2) We return a `{sheet_name: frame,}` dict when loading multiple sheets - there isn't an especially clear way to indicate that you want to merge the results from this dict when additionally loading from multiple workbooks.

So, the new option covers the more common use-cases clearly - if somebody has the use-case you describe then it isn't hard to construct, but I suspect it is relatively uncommon.

Reply:
Thanks. (The use-case which motivated my feature request is exactly this use-case. 😄)

I imagine there are a few possible ways you could do this.

1. Require all files to have the same list of sheet names, and all matching sheets to have the same list of field names. Else error.
2. Require all _matching_ sheets to have the same list of field names, but allow different lists of sheets in each file. E.g: File 1 has sheets `A, B, C`. File 2 has sheets `B, D`. Sheet `B` in both files has matching fields. Resulting dict is the union of all sheets, with matching sheets concatenated: `{'A': ..., 'B': ..., 'C': ..., 'D': ...}`. Error if sheets with matching names cannot be concatenated.

Option 1. is a subset of option 2.

Option 1. is sufficient for my use case, and perhaps less confusing.

But I kind of like the generality of option 2.

Reply:
Yup, I was tending towards option 2 as well...
I'll see if I can add it in the next day or two, festivities not withstanding 😉👍

Reply:
Done: #20465.

Reply:
Amazing! Thanks!